### PR TITLE
feat(calendar): add value type String

### PR DESCRIPTION
**Pull request by Tvinsh**
change for v-model and range to support String type, change docs for new feature, and remove no use
test case

- [ calendar] Calendar add some feature

### CODE REVIEW

File: packages/calendar/__tests__/calendar.spec.ts [hunk 1]
```diff
@@ -61,29 +72,4 @@ describe('Calendar.vue', () => {
     expect(/2019.*May/.test((titleEl.element as HTMLElement).innerHTML)).toBeTruthy()
     expect(cell.classList.contains('is-selected')).toBeTruthy()
   })
-
```
Comment (zazzaz) [inline packages/calendar/__tests__/calendar.spec.ts:64]: Any reason we should delete this test?

**Comment by zazzaz** [comment]
Thanks for the update of the docs.

IMHO, components that are related to date and time, should accept and emit native Date type only. In this way, we can make the best compatibility among all browsers.

File: packages/calendar/__tests__/calendar.spec.ts (hunk 1 above)
Comment (Tvinsh) [reply packages/calendar/__tests__/calendar.spec.ts:64]: firstDayOfWeek prop is no use in calendar code

**Review by Tvinsh — COMMENTED** [review]
> Thanks for updating the docs.
> 
> IMHO, components that are related to date and time, should accept and emit native Date type only. In this way, we can make the best compatibility among all browsers.

so what problems may string value ​​cause

**Comment by zazzaz** [comment]
> so what problems may string value ​​cause

The problem is not the string value itself.  The date object is a better way to dementrating date and time, it contains more information that a simple date string like uct and offset.

File: packages/calendar/__tests__/calendar.spec.ts (hunk 1 above)
Comment (zazzaz) [reply packages/calendar/__tests__/calendar.spec.ts:64]: Although it's not used in the code, firstDayOfWeek is still a valid setting configed by dayjs.locale() in the global scope. 

The test is still needed.

**Comment by Tvinsh** [comment]
> > so what problems may string value ​​cause
> 
> The problem is not the string value itself. The date object is a better way to dementrating date and time, it contains more information that a simple date string like uct and offset.

Thanks for replying, I think this does not affect the string value, it can reduce the breaking change, if you do not think this,i will close this pr

**Comment by zazzaz** [comment]
> > > so what problems may string value ​​cause
> > 
> > 
> > The problem is not the string value itself. The date object is a better way to dementrating date and time, it contains more information that a simple date string like uct and offset.
> 
> Thanks for replying, I think this does not affect the string value, it can reduce the breaking change, if you do not think this,i will close this pr

We can keep the part related to the doc update.

**Comment by JeremyWuuuuu** [comment]
> > so what problems may string value ​​cause
> 
> The problem is not the string value itself. The date object is a better way to dementrating date and time, it contains more information that a simple date string like uct and offset.

Agreed, Date Objects are better for containing extra informations but string value is not.

**Comment by JeremyWuuuuu** [comment]
@Tvinsh Could you please revert the part of code which made changes towards the type? We can still use the documenation part.

**Comment by Tvinsh** [comment]
@zazzaz @JeremyWuuuuu Thanks for replying, change is revert

**Comment by zazzaz** [comment]
Cool, thanks.

**Pull request merged**